apps/server/src/shared/domain/service/permission.service.ts

```diff
@@ -53,10 +61,95 @@ export class PermissionService {
 		return hasPermissions;
 	}
 
-	checkUserHasAllSchoolPermissions(user: User, requiredPermissions: string[]): void {
-		const hasPermission = this.hasUserAllSchoolPermissions(user, requiredPermissions);
+	/**
+	 * Determines whether a user has all the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @throws UnauthorizedException
+	 */
+	checkUserHasAllPermissions(user: User, requiredPermissions: string[]): void {
+		const hasPermission = this.hasUserAllPermissions(user, requiredPermissions);
+		if (hasPermission !== true) {
+			throw new UnauthorizedException();
+		}
+	}
+
+	/**
+	 * Determines whether a user has at least one of the rights specified in requiredPermissions
+	 * @param user
+	 * @param requiredPermissions
+	 * @returns boolean
+	 */
+	hasUserOneOfPermissions(user: User, requiredPermissions: string[]): boolean {
```

Comment:
New pattern like OneOf or HasAll is ok. But it is also possible to re-use the array js names for the similar operations. That are know by each developer. Some / Every (each)